### ISSUE TITLE
AttributeError: 'NodeDictClass' object has no attribute 'x'

### ISSUE THREAD

**Issue by WiseLord** [PiotrMachowski/Python-package-vacuum-map-parser-base#7]
Hello.

I was experimenting with [texts in Xiaomi-Cloud-Map-Extractor](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/tree/dev_extracted_libraries?tab=readme-ov-file#texts-configuration) and got the error in this module.

```
Dec 15 00:23:05 bpim5 hass[68285]: Traceback (most recent call last):
Dec 15 00:23:05 bpim5 hass[68285]:   File "/srv/homeassistant/lib/python3.13/site-packages/homeassistant/helpers/entity.py", line 960, in async_update_ha_state
Dec 15 00:23:05 bpim5 hass[68285]:     await self.async_device_update()
Dec 15 00:23:05 bpim5 hass[68285]:   File "/srv/homeassistant/lib/python3.13/site-packages/homeassistant/helpers/entity.py", line 1320, in async_device_update
Dec 15 00:23:05 bpim5 hass[68285]:     await hass.async_add_executor_job(self.update)
Dec 15 00:23:05 bpim5 hass[68285]:   File "/usr/local/lib/python3.13/concurrent/futures/thread.py", line 59, in run
Dec 15 00:23:05 bpim5 hass[68285]:     result = self.fn(*self.args, **self.kwargs)
Dec 15 00:23:05 bpim5 hass[68285]:   File "/home/hass/.homeassistant/custom_components/xiaomi_cloud_map_extractor/camera.py", line 281, in update
Dec 15 00:23:05 bpim5 hass[68285]:     self._render_map()
Dec 15 00:23:05 bpim5 hass[68285]:     ~~~~~~~~~~~~~~~~^^
Dec 15 00:23:05 bpim5 hass[68285]:   File "/home/hass/.homeassistant/custom_components/xiaomi_cloud_map_extractor/camera.py", line 314, in _render_map
Dec 15 00:23:05 bpim5 hass[68285]:     self._image_generator.draw_map(self._map_data)
Dec 15 00:23:05 bpim5 hass[68285]:     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^
Dec 15 00:23:05 bpim5 hass[68285]:   File "/srv/homeassistant/lib/python3.13/site-packages/vacuum_map_parser_base/image_generator.py", line 79, in draw_map
Dec 15 00:23:05 bpim5 hass[68285]:     self._draw_texts(map_data.image, self._texts)
Dec 15 00:23:05 bpim5 hass[68285]:     ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Dec 15 00:23:05 bpim5 hass[68285]:   File "/srv/homeassistant/lib/python3.13/site-packages/vacuum_map_parser_base/image_generator.py", line 252, in _draw_texts
Dec 15 00:23:05 bpim5 hass[68285]:     x = text_config.x * image.data.size[0] / 100
Dec 15 00:23:05 bpim5 hass[68285]:         ^^^^^^^^^^^^^
Dec 15 00:23:05 bpim5 hass[68285]: AttributeError: 'NodeDictClass' object has no attribute 'x'
```

I rewrote `_draw_texts()` in `/srv/homeassistant/lib/python3.13/site-packages/vacuum_map_parser_base/image_generator.py` (I use "core" installation using venv)

from:

```
    @staticmethod
    def _draw_texts(image: ImageData, texts: list[Text]) -> None:
        for text_config in texts:
            x = text_config.x * image.data.size[0] / 100
            y = text_config.y * image.data.size[1] / 100
            ImageGenerator._draw_text(
                image,
                text_config.text,
                x,
                y,
                text_config.color,
                text_config.font,
                text_config.font_size,
            )
```

to:

```
    @staticmethod
    def _draw_texts(image: ImageData, texts: list[Text]) -> None:
        for text_config in texts:
            x = text_config['x'] * image.data.size[0] / 100
            y = text_config['y'] * image.data.size[1] / 100
            ImageGenerator._draw_text(
                image,
                text_config['text'],
                x,
                y,
                text_config['color'],
                text_config['font'],
                text_config['font_size'],
            )

```

and the error disappeared and my "camera" started rendering the text.

What it could be? Some changes in new python version (my HA instance uses 3.13.1 for now)?

That was my config that started causing the error when I added `text` into it:

```

camera:
  - platform: xiaomi_cloud_map_extractor
    host: !secret xiaomi_vacuum_host
    token: !secret xiaomi_vacuum_token
    username: !secret xiaomi_cloud_username
    password: !secret xiaomi_cloud_password
    store_map_raw: true
    store_map_image: true
    store_map_path: /tmp
    draw:
      - charger
      - cleaned_area
      - goto_path
      - ignored_obstacles
      - ignored_obstacles_with_photo
      - no_go_zones
      - no_mopping_zones
      - obstacles
      - obstacles_with_photo
      - path
      - predicted_path
      # - room_names
      - vacuum_position
      - virtual_walls
      - zones
    texts:
      - text: "Room 2"
        x: 45
        y: 45
        color: [125, 20, 213, 127]
        font: "LiberationSans-Regular.ttf"
        font_size: 25
    attributes:
      - calibration_points
    map_transformation:
      scale: 2
      rotate: 0
      trim:
        top: 44
        bottom: 32
        left: 39
        right: 25
```

**Comment by PiotrMachowski** [comment]
Hi, was this config working before updating to Python 3.13?

The problem is on the side of Map Extractor - it sends an incorrect type of data to the map parser. I have fixed these things, but I haven't uploaded these changes to the repo.

Required changes to be made [here](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/blob/c5bcfc213c06841661a637ae639c60dd8bd494ad/custom_components/xiaomi_cloud_map_extractor/camera.py#L133):
```python
    transform = config[CONF_MAP_TRANSFORM]
    image_config = ImageConfig(**{**transform, "trim": TrimConfig(**transform["trim"])})
    colors = ColorsPalette(config[CONF_COLORS], {str(k): v for k, v in config[CONF_ROOM_COLORS].items()})
    drawables = config[CONF_DRAW]
    sizes = Sizes(config[CONF_SIZES])
    texts = [Text(**t) for t in config[CONF_TEXTS]]
```

Additionally import in [line 8](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/blob/c5bcfc213c06841661a637ae639c60dd8bd494ad/custom_components/xiaomi_cloud_map_extractor/camera.py#L8) has to be adjusted:
```python
from vacuum_map_parser_base.config.image_config import ImageConfig, TrimConfig
```

**Comment by WiseLord** [comment]
> Hi, was this config working before updating to Python 3.13?

I don't know, actually. I started experimenting only few days ago, already being with this version.

Also, I used the [fork of the fork](https://github.com/Nicooow/HA-custom-components-Xiaomi-Cloud-Map-Extractor-Ijai-Support/tree/ijai_map_image) of your project that seems to support my Xiaomi Robot Vacuum S10 (ijai.vacuum.v17).

The link to this fork I found in the discussion on the [issue 460](https://github.com/PiotrMachowski/Home-Assistant-custom-components-Xiaomi-Cloud-Map-Extractor/issues/460)

Anyway, thanks for the answer. I'll try the fix on extractor side with non-modified base parser.
